File: sort/选择排序/section.go
package main

import "fmt"

func main() {
	number := []int{95, 45, 15, 78, 84, 51, 24, 12}
	fmt.Println("Sort Result:", SectionSort(number))
}

/*
	SectionSort 选择排序
	input: [95 45 15 78 84 51 24 12]
	output: [12 15 24 45 51 78 84 95]
*/
func SectionSort(values []int) []int {
	length := len(values)
	for i := 0; i < length-1; i++ {
		min := i
		for j := i + 1; j < length; j++ {
			if values[j] < values[min] {
				min = j
			}
		}
		values[i], values[min] = values[min], values[i]
	}
	return values
}

File: sort/选择排序/section.py
# input: [4, 1, 5, 2, 3]
# output: [1, 2, 3, 4, 5]

def sectionSort(nums):
    for i in range(len(nums)-1):
        min = i
        for j in range(i+1,len(nums)):
            if nums[j] < nums[min]:
                min = j
        nums[i], nums[min] = nums[min], nums[i]
    return nums

nums = [4,1,5,2,3]
print "input:",nums
print "output:",sectionSort(nums)